google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/DeploymentArtifactType.java

```diff
@@ -31,11 +32,13 @@
    * Returns the right {@code DeploymentArtifactType} for the given {@code deployPackage}.
    */
   @NotNull
-  public static DeploymentArtifactType typeForPath(@NotNull File deployPackage) {
-    if (deployPackage.getPath().endsWith(".jar")) {
-      return JAR;
-    } else if (deployPackage.getPath().endsWith(".war")) {
-      return WAR;
+  public static DeploymentArtifactType typeForPath(@Nullable File deployPackage) {
```

Comment:
apologies for flip flopping, but couldn't we leave this as @NotNull and just have the callers of this method handle the case where deploymenttSource.getFile == null?